### PR TITLE
Login with instagram

### PR DESCRIPTION
**Pull request by SaenkoM**

### CODE REVIEW

File: src/js/components/SignIn.jsx [hunk 1]
```diff
@@ -1,128 +1,130 @@
...
-
-				<button disabled={isInvalid} type="submit" className="button cta">
-					Sign In
-				</button>
-
-				{/* <button
+  constructor(props) {
+    super(props);
+
+    this.state = { ...INITIAL_STATE };
```
Comment (SaenkoM) [inline src/js/components/SignIn.jsx]: `state= INITIAL_STATE` outside of constructor would do the same

File: src/js/components/SignIn.jsx [hunk 2]
```diff
@@ -1,128 +1,130 @@
...
+      });
+
+    event.preventDefault();
+  };
+
+  onChange = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  // facebookSignIn = event => {
```
Comment (SaenkoM) [inline src/js/components/SignIn.jsx]: Please uncomment or delete.

File: src/js/components/SocialLogin.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  render() {
+    return (
+      <div className="social-login-container">
+        <p className="social-login-title">Or log in with social:</p>
+        <div className="social-login-icons">
+          <div className="social-login-icons-container">
+            <a
+              href="https://api.instagram.com/oauth/authorize/?client_id=2fd5e7a0646f4f9b97367cc42f5c246a&redirect_uri=http://localhost:3000/&response_type=token"
```
Comment (SaenkoM) [inline src/js/components/SocialLogin.jsx]: Should be a constant, `client_id` should be in .env.

File: src/js/containers/LandingPage.jsx [hunk 4]
```diff
@@ -1,157 +1,172 @@
...
-			window.ga('gtag_UA_140019312_1.set', 'page', '/');
-			window.ga('gtag_UA_140019312_1.send', 'pageview');
-		}
+  async componentDidMount() {
+    //------------------------------------------------------------------------------
+    console.log("check url", window.location.href);
+    let url = window.location.href;
+    const access_token = url.split("access_token=")[1];
+    console.log("access token", access_token);
+    this.props.firebase.doSignInWithInstagram(access_token);
```
Comment (SaenkoM) [inline src/js/containers/LandingPage.jsx]: `firebase` is used multiple times, prefer object destructuring even if used one time.

File: src/js/containers/LandingPage.jsx [hunk 5]
```diff
@@ -1,157 +1,172 @@
...
-				this.setState({
-					dateLastCheckIn: JSON.parse(
-						localStorage.getItem('assessment.dateLastCheckIn')
-					)
-				});
-			}
-		}
+    if (this.props.firebase) {
+      this.listener = this.props.firebase.auth.onAuthStateChanged(authUser => {
+        authUser
```
Comment (SaenkoM) [inline src/js/containers/LandingPage.jsx]:
```
this.setState({
  authUser: authUser || null,
})
```

File: src/js/containers/LandingPage.jsx [hunk 6]
```diff
@@ -1,157 +1,172 @@
...
+            })
+          : this.setState({
+              authUser: null
+            });
+      });
+    }
 
-		this.getLatestGoalOfTheDayEntry();
-	}
+    if (typeof window !== "undefined") {
```
Comment (SaenkoM) [inline src/js/containers/LandingPage.jsx]: Why window would be undefined here?

File: src/js/containers/LandingPage.jsx [hunk 7]
```diff
@@ -1,157 +1,172 @@
...
-	checkInRouting = () => {
-		const { dateLastCheckIn } = this.state;
+    client
+      .getEntries({
+        content_type: "goalOfTheDay",
+        order: "-sys.updatedAt",
+        limit: 1
+      })
+      .then(entry => {
+        if (entry.items.length > 0) {
```
Comment (SaenkoM) [inline src/js/containers/LandingPage.jsx]: `entry.items.length` is enough

File: src/js/containers/LandingPage.jsx [hunk 8]
```diff
@@ -1,157 +1,172 @@
...
+          this.setState({
+            gotdValue: entry.items[0].fields.questionValue
```
Comment (SaenkoM) [inline src/js/containers/LandingPage.jsx]: Looks like a very long nesting, something can go wrong. You can safely access questionValue by using lodash's `get`:
```
import _get from 'lodash/get
const gotdValue = _get(entry, ['items', 0, 'fields', 'questionValue'], defaultValueIfNeeded)
```

File: src/js/containers/LandingPage.jsx (hunk 6 above)
Comment (danispavel) [reply src/js/containers/LandingPage.jsx]: I don't know it's not my part of code

**Pull request merged**